CoreQualityControlView/test/unit/src/au/gov/asd/tac/constellation/views/qualitycontrol/widget/DefaultQualityControlAutoButtonNGTest.java

```diff
@@ -0,0 +1,135 @@
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+public class DefaultQualityControlAutoButtonNGTest {
+
+    private StoreGraph graph;
+    private List<QualityControlEvent> events;
+    private List<QualityControlRule> rules;
+
+    //Dependencies (will be mocked)
+    private QualityControlEvent qualityControlEvent;
+
+    public DefaultQualityControlAutoButtonNGTest() {
+    }
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+    }
+
+    @BeforeMethod
+    public void setUpMethod() throws Exception {
+        final Schema schema = SchemaFactoryUtilities.getSchemaFactory(AnalyticSchemaFactory.ANALYTIC_SCHEMA_ID).createSchema();
+        graph = new StoreGraph(schema);
+        rules = new ArrayList<>();
+        events = new ArrayList<>();
+
+        //mock QualityControlEvent
+        qualityControlEvent = mock(QualityControlEvent.class);
+        when(qualityControlEvent.getCategory()).thenReturn(QualityCategory.FATAL);
+        when(qualityControlEvent.getReasons()).thenReturn("Reason 1, Reason2");
+
+        events.add(qualityControlEvent);
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+    }
+
+    /**
+     * Test of qualityControlChanged method, of class
+     * DefaultQualityControlAutoButton, with Null State.
+     */
+    @Test
+    public void testQualityControlChangedWithNullState() throws InterruptedException {
+        System.out.println("qualityControlChanged");
+        QualityControlState state = null;
+        if (!GraphicsEnvironment.isHeadless()) {
+            //needed to initialise the toolkit
+            new JFXPanel();
+            DefaultQualityControlAutoButton instance = new DefaultQualityControlAutoButton();
+
+            String expRiskText = String.format(instance.QUALITY_CONTROL_WIDGET_TEXT, Bundle.MSG_NoRisk());
+            String expStyleText = instance.DEFAULT_TEXT_STYLE + instance.BUTTON_STYLE;
+            String expTooltipText = null;
+
+            instance.qualityControlChanged(state);
+
+            String resultRiskText = instance.getText();
+            String resultStyleText = instance.getStyle();
+            final Tooltip resultTooltipText = instance.getTooltip();
+
+            assertEquals(resultRiskText, expRiskText);
+            assertEquals(resultStyleText, expStyleText);
+            assertEquals(resultTooltipText, expTooltipText);
+        }
+    }
+
+    /**
+     * Test of qualityControlChanged method, of class
+     * DefaultQualityControlAutoButton, with a valid State.
+     */
+    @Test
+    public void testQualityControlChangedWithValidState() throws InterruptedException {
+        System.out.println("qualityControlChanged");
+        QualityControlState state = new QualityControlState(graph.getId(), events, rules);
```

Comment:
Since this state won't be used if the subsequent if condition is false, I would move this to inside the if block